File: ruby/lib/enolib/elements/missing/missing_empty.rb
# frozen_string_literal: true

module Enolib
  class MissingEmpty < MissingElementBase
    def to_s
      if @key
        "#<Enolib::MissingEmpty key=#{@key}>"
      else
        '#<Enolib::MissingEmpty>'
      end
    end
  end
end

File: javascript/lib/elements/missing/missing_empty.js
const { MissingElementBase } = require('./missing_element_base.js');

class MissingEmpty extends MissingElementBase {
  get [Symbol.toStringTag]() {
    return 'MissingEmpty';
  }

  toString() {
    if(this._key === null)
      return `[object MissingEmpty]`;

    return `[object MissingEmpty key=${this._key}]`;
  }
}

exports.MissingEmpty = MissingEmpty;
